tools/gfx/d3d12/render-d3d12.cpp

```diff
@@ -1797,8 +1886,34 @@ class D3D12Device : public RendererBase
                 D3D12_DESCRIPTOR_RANGE_TYPE rangeType,
                 UINT registerIndex,
                 UINT spaceIndex,
-                UINT count)
+                UINT count,
+                bool isRootParameter)
             {
+                if (isRootParameter)
+                {
+                    D3D12_ROOT_PARAMETER rootParam = {};
+                    switch (rangeType)
+                    {
+                    case D3D12_DESCRIPTOR_RANGE_TYPE_SRV:
+                        rootParam.ParameterType = D3D12_ROOT_PARAMETER_TYPE_SRV;
+                        break;
+                    case D3D12_DESCRIPTOR_RANGE_TYPE_UAV:
+                        rootParam.ParameterType = D3D12_ROOT_PARAMETER_TYPE_UAV;
+                        break;
+                    default:
+                        getDebugCallback()->handleMessage(
+                            DebugMessageType::Error,
+                            DebugMessageSource::Layer,
+                            "A shader parameter marked as root parameter is neither SRV nor UAV.");
+                        return SLANG_FAIL;
+                    }
```

Comment:
Doesn't D3D12 support binding constant buffers as root parameters as well?

Reply:
They do, but supporting it in gfx requires some more careful handling because they are ShaderObjects instead of a plain resource, so I leave it unimplemented for now.